invoice/src/main/java/org/killbill/billing/invoice/model/DefaultInvoice.java

```diff
@@ -249,7 +249,9 @@ public BigDecimal getRefundedAmount() {
 
     @Override
     public BigDecimal getBalance() {
-        return hasZeroParentBalance() ? BigDecimal.ZERO : InvoiceCalculatorUtils.computeInvoiceBalance(currency, invoiceItems, payments, isWrittenOff() || isMigrationInvoice());
+        return getStatus().equals(InvoiceStatus.DRAFT) || hasZeroParentBalance() ?
+               BigDecimal.ZERO :
+               InvoiceCalculatorUtils.computeInvoiceBalance(currency, invoiceItems, payments, isWrittenOff() || isMigrationInvoice());
     }
```

Comment:
Should we only ask `getStatus().equals(InvoiceStatus.DRAFT)` for parent invoices? Current code doesn't matter what kind of invoice is.